virtool/samples/data.py

```diff
@@ -158,23 +159,28 @@ async def find(
         )
 
     async def get(self, sample_id: str) -> Sample:
-        document = await self._db.samples.find_one({"_id": sample_id})
+        documents = await self._db.samples.aggregate(
+            [
+                {"$match": {"_id": sample_id}},
+                *lookup_nested_user_by_id(local_field="user.id"),
+                *lookup_nested_subtractions(local_field="subtractions"),
+                *lookup_caches(local_field="_id"),
+            ]
+        ).to_list(length=1)
 
-        if not document:
-            raise ResourceNotFoundError
+        document = None
 
-        document["caches"] = [
-            base_processor(cache)
-            async for cache in self._db.caches.find({"sample.id": sample_id})
-        ]
+        if len(documents) == 0:
+            raise ResourceNotFoundError
+        else:
+            document = documents[0]
+            del documents
```

Comment:
No need for the `else`, if `len(documents) == 0` then an exception will be raised preventing execution of the rest of the function. This also lets you avoid pre-declaring `document` as `None`